src/assert_output.bash

```diff
@@ -187,11 +192,17 @@ assert_output() {
     fi
   else
     if [[ $output != "$expected" ]]; then
-      batslib_print_kv_single_or_multi 8 \
-      'expected' "$expected" \
-      'actual'   "$output" \
-      | batslib_decorate 'output differs' \
-      | fail
+      if (( show_diff )); then
+        diff <(echo "$expected") <(echo "$output") \
```

Comment:
The _unified_ file format is more common than the so called _normal_ format. It would be better to use `diff -u` instead of just `diff` here.